returnn/util/file_cache.py

```diff
@@ -327,6 +327,20 @@ def _check_existing_copied_file_maybe_cleanup(src_filename: str, dst_filename: s
         return True
 
 
+def _copy_with_prealloc(src: str, dst: str):
+    """
+    copies from src to dst preallocating the disk space at dst before copying
+    :param src: source file
+    :param dst: destination file
+    """
+    stat = os.stat(src)
+    with open(dst, "wb") as dst_file:
```

Comment:
I think calling `os.posix_fadvise` with `POSIX_FADV_SEQUENTIAL | POSIX_FADV_NOREUSE` or so might also be a good idea?

Reply:
Note: I did not set `POSIX_FADV_NOREUSE` because we actually access the file range targeted by the advise multiple times (once per chunk copied). Alternative would be one advise call per chunk, but this seems like a lot of syscall overhead and also would require us to stop using `copyfileobj` and hand-roll the copying loop, which isn't worth it.

Reply:
But my understanding of `POSIX_FADV_NOREUSE` is that it is per byte, not per range. You just specify the range to avoid calling `fadvise` size-times. So it's actually exactly fitting for our use case.

Otherwise, `POSIX_FADV_NOREUSE` would not really make sense to me this way. I think almost all existing file copy code (e.g. check `/bin/cp`) use exactly this logic. (But we can actually check that...)

Reply:
Ok, in GNU coretuils, there is a comment saying `POSIX_FADV_NOREUSE` does nothing since Linux kernel version 2.6.31?
https://github.com/coreutils/coreutils/blob/bbc972bc827aa5b34fce697ed79a36c70c4fabe3/src/sort.c#L936-L941

Ok, so then maybe we can really ignore this...